test/cmd/certs.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+trap os::test::junit::reconcile_output EXIT
+
+os::test::junit::declare_suite_start "cmd/admin/certs-validation"
+
+CERT_DIR="${BASETMPDIR:-/tmp}/certs"
+mkdir -p "${CERT_DIR}"
+
+# oadm ca create-signer-cert should generate certificate for 5 years by default
+os::cmd::expect_success_and_not_text \
+    "oadm ca create-signer-cert --cert=${CERT_DIR}/ca.crt --key=${CERT_DIR}/ca.key --serial=${CERT_DIR}/ca.serial.txt --overwrite=true" \
+    'WARNING: .* is greater than 5 years'
+
+expected_year="$(TZ=GMT date -d "+$[365*5] days" +'%Y')"
+
+os::cmd::expect_success_and_text "openssl x509 -in ${CERT_DIR}/ca.crt -enddate -noout | cut -d' ' -f4" "${expected_year}"
+
+# oadm ca create-signer-cert should generate certificate with specified number of days and show warning
+os::cmd::expect_success_and_text \
+    "oadm ca create-signer-cert --cert=${CERT_DIR}/ca.crt --key=${CERT_DIR}/ca.key --serial=${CERT_DIR}/ca.serial.txt --overwrite=true --expire-days=$[365*6]" \
+    'WARNING: .* is greater than 5 years'
+
+expected_year="$(TZ=GMT date -d "+$[365*6] days" +'%Y')"
```

Comment:
>If these tests ever run close to midnight on Dec 31, they will fail, right?

It's very very rarely case. It means that test is executing in 23:59:59 on Dec 31 and between calculation of the expected year and creation of the certificate gone >1 second.

>we don't really need to check that the utility that we use to create the cert is correctly using the end-date value we pass it, do we?

Actually it was my intention. I want to be sure that this option works as expected on the user level.